decred/tests/unit/dcr/test_vsp_unit.py

```diff
@@ -0,0 +1,354 @@
+    b = vsp.VotingServiceProvider.blob(pool)
+    assert isinstance(b, bytearray)
+
+    rePool = vsp.VotingServiceProvider.unblob(b)
+    assert rePool.url == "https://www.dcrstakedinner.com"
+    assert rePool.apiKey == votingServiceProvider["apiKey"]
+    assert rePool.net.Name == "testnet3"
+    assert rePool.purchaseInfo.ticketAddress == "Tcbvn2hiEAXBDwUPDLDG2SxF9iANMKhdVev"
+
+    # bad version
+    bCopy = encode.ByteArray(b, copy=True)
+    bCopy[0] = 255
+    with pytest.raises(Exception):
+        vsp.VotingServiceProvider.unblob(bCopy.bytes())
+
+    # too long
+    bCopy = encode.ByteArray(b, copy=True)
+    bCopy += b"\x00"
+    with pytest.raises(Exception):
+        vsp.VotingServiceProvider.unblob(bCopy.bytes())
+
+
+def test_vsp_serialize():
+    pool = vsp.VotingServiceProvider(**votingServiceProvider)
+    b = vsp.VotingServiceProvider.blob(pool)
+    assert pool.serialize() == encode.ByteArray(b)
+
+
+# fmt: off
+vspProviders = {'Staked': {'APIEnabled': True, 'APIVersionsSupported': [1, 2], 'Network': 'mainnet', 'URL': 'https://decred.staked.us', 'Launched': 1543433400, 'LastUpdated': 1582020568, 'Immature': 0, 'Live': 141, 'Voted': 2730, 'Missed': 10, 'PoolFees': 5, 'ProportionLive': 0.0034847511245118877, 'ProportionMissed': 0.0036496350364963502, 'UserCount': 229, 'UserCountActive': 106, 'Version': '1.4.0-pre+dev'}, 'Golf': {'APIEnabled': True, 'APIVersionsSupported': [1, 2], 'Network': 'mainnet', 'URL': 'https://stakepool.dcrstats.com', 'Launched': 1464167340, 'LastUpdated': 1582020568, 'Immature': 21, 'Live': 768, 'Voted': 148202, 'Missed': 154, 'PoolFees': 5, 'ProportionLive': 0.01898077208244773, 'ProportionMissed': 0, 'UserCount': 6005, 'UserCountActive': 2751, 'Version': '1.5.0-pre'}, 'Hotel': {'APIEnabled': True, 'APIVersionsSupported': [1, 2], 'Network': 'mainnet', 'URL': 'https://stake.decredbrasil.com', 'Launched': 1464463860, 'LastUpdated': 1582020568, 'Immature': 41, 'Live': 607, 'Voted': 48135, 'Missed': 49, 'PoolFees': 5, 'ProportionLive': 0.015002842383647644, 'ProportionMissed': 0.0010169350821849577, 'UserCount': 1607, 'UserCountActive': 968, 'Version': '1.5.0'}, 'November': {'APIEnabled': True, 'APIVersionsSupported': [1, 2], 'Network': 'mainnet', 'URL': 'https://decred.raqamiya.net', 'Launched': 1513878600, 'LastUpdated': 1582020568, 'Immature': 5, 'Live': 334, 'Voted': 15720, 'Missed': 50, 'PoolFees': 1, 'ProportionLive': 0.008255270767937913, 'ProportionMissed': 0.0031705770450221942, 'UserCount': 261, 'UserCountActive': 114, 'Version': '1.5.0-pre'}, 'Ray': {'APIEnabled': True, 'APIVersionsSupported': [1, 2], 'Network': 'mainnet', 'URL': 'https://dcrpos.idcray.com', 'Launched': 1518446640, 'LastUpdated': 1582020569, 'Immature': 50, 'Live': 1108, 'Voted': 36974, 'Missed': 298, 'PoolFees': 2, 'ProportionLive': 0.027385748535554512, 'ProportionMissed': 0.007995277956643057, 'UserCount': 137, 'UserCountActive': 70, 'Version': '1.4.0-pre+dev'}}
+# fmt: on
```

Comment:
Is it really necessary to have one very long line here? It's unreadable and undiffable. You may also use the `dict()` constructor to avoid string keys quoting.

Reply:
Copy-paste the output from `json.dumps(json.loads(vspProviders, indent=4, sort_keys=True))`.